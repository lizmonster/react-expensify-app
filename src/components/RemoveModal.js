import React from 'react';
import Modal from 'react-modal';

const OptionModal = (props) => (
    <Modal
        isOpen={!!props.removingExpense}
        ariaHideApp={false}
        contentLabel="Remove Expense"
        closeTimeoutMS={
            200
        }
        className="modal"
    >
        <h3 className="modal__title">Remove Expense</h3>
        <h3 className="modal__name">"{props.expenseName}" ?</h3>
        <button className="button modal__button" onClick={props.handleRemove}>Yes</button>
        <button className="button modal__button" onClick={props.onDismiss}>No</button>
    </Modal>
);

export default OptionModal;