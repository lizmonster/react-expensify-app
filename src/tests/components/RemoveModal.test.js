import React from 'react';
import { shallow } from 'enzyme';
import RemoveModal from '../../components/RemoveModal';

test('should render RemoveModal dialog correctly', () => {
    const wrapper = shallow(<RemoveModal />);
    expect(wrapper).toMatchSnapshot();
});