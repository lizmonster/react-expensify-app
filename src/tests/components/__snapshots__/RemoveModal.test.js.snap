// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render RemoveModal dialog correctly 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="Remove Expense"
  isOpen={false}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h3
    className="modal__title"
  >
    Remove Expense
  </h3>
  <h3
    className="modal__name"
  >
    "
    " ?
  </h3>
  <button
    className="button modal__button"
  >
    Yes
  </button>
  <button
    className="button modal__button"
  >
    No
  </button>
</Modal>
`;
